ca/ca.go

```diff
@@ -306,7 +306,7 @@ func (ca *certificateAuthorityImpl) IssuePrecertificate(ctx context.Context, iss
 		return nil, err
 	}
 
-	precertDER, certProfileName, certProfileHash, err := ca.issuePrecertificateInner(ctx, issueReq, serialBigInt, validity)
+	precertDER, certProfileWithID, err := ca.issuePrecertificateInner(ctx, issueReq, serialBigInt, validity)
```

Comment:
naming this variable `certProfileWithID` "shadows" the type name `certProfileWithID`, making it impossible for any code below this point in the function to construct a `certProfileWithID` struct. There's no reason for code below here to want to, but it's still best practice not to shadow.
```suggestion
	precertDER, usedProfile, err := ca.issuePrecertificateInner(ctx, issueReq, serialBigInt, validity)
```

Reply:
I didn't realize that was possible, but it totally makes sense. Good to know, thank you.